src/onyx/peer/task_lifecycle.clj

```diff
@@ -203,7 +203,7 @@
           metadata-bs (checkpoint-compress {:state-indexes (ws/state-indices event)})
           _ (cpenc/set-schema-version checkpoint-encoder 0)
           _ (cpenc/set-metadata checkpoint-encoder metadata-bs)
-          exported-state (db/export (get-state-store state) checkpoint-encoder) 
+          _ (db/export (get-state-store state) checkpoint-encoder) 
```

Comment:
Intentional/necessary? Is this side-effecting?

Reply:
Yes, it serializes it into the encoder, so the old binding was incorrect.